### PR TITLE
tiflash: Add TiFlash security configuration

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Signed-off-by: JaySon-Huang <tshent@qq.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add document for security configuration in TiFlash.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/5094
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by JaySon-Huang** [comment]
@flowbehappy

File: tiflash/tiflash-configuration.md [hunk 1]
```diff
@@ -90,6 +90,22 @@ Multiple TiFlash nodes elect a master to add or delete placement rules to PD, an
     dt_enable_logical_split = true # The default value is `true`. This parameter determines whether the segment of DeltaTree Storage Engine uses logical split. Using the logical split can reduce the write amplification, and improve the write speed. However, these are at the cost of disk space waste.
     max_memory_usage = 0 # The memory usage limit for the generated intermediate data when a single coprocessor query is executed. The default value is 0, which means no limit.
     max_memory_usage_for_all_queries = 0 # The memory usage limit for the generated intermediate data when all queries are executed. The default value is 0 (in bytes), which means no limit.
+
+## Security settings effective since v4.0.5
```
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:94]:
```suggestion
## Security settings take effect starting from v4.0.5.
```

File: tiflash/tiflash-configuration.md [hunk 2]
```diff
@@ -90,6 +90,22 @@ Multiple TiFlash nodes elect a master to add or delete placement rules to PD, an
...
+[security]
+    ## This configuration item enables or disables log redaction. If the configuration value
+    ## is set to `true`, all user data in the log will be replaced by `?`. Also, you need to set
+    ## `security.redact-info-log` in tiflash-learner.toml to enable it for tiflash-learner
+    ## logging files.
```
Comment (JaySon-Huang) [inline tiflash/tiflash-configuration.md:99]:
```suggestion
    ## This configuration item enables or disables log redaction. If the configuration value
    ## is set to `true`, all user data in the log will be replaced by `?`.
    ## Note that you need to enable `security.redact-info-log` for tiflash-learner's logging
    ## in tiflash-learner.toml
    ## logging files.
```

File: tiflash/tiflash-configuration.md [hunk 3]
```diff
@@ -90,6 +90,23 @@ Multiple TiFlash nodes elect a master to add or delete placement rules to PD, an
     dt_enable_logical_split = true # The default value is `true`. This parameter determines whether the segment of DeltaTree Storage Engine uses logical split. Using the logical split can reduce the write amplification, and improve the write speed. However, these are at the cost of disk space waste.
     max_memory_usage = 0 # The memory usage limit for the generated intermediate data when a single coprocessor query is executed. The default value is 0, which means no limit.
     max_memory_usage_for_all_queries = 0 # The memory usage limit for the generated intermediate data when all queries are executed. The default value is 0 (in bytes), which means no limit.
+
+## Security settings take effect starting from v4.0.5.
+[security]
+    ## This configuration item enables or disables log redaction. If the configuration value
+    ## is set to `true`, all user data in the log will be replaced by `?`.
+    ## Note that you need to enable `security.redact-info-log` for tiflash-learner's logging
+    ## in tiflash-learner.toml
```
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:99]:
```suggestion
    ## Note that you also need to set `security.redact-info-log` for tiflash-learner's logging
    ## in tiflash-learner.toml
```

File: tiflash/tiflash-configuration.md [hunk 4]
```diff
@@ -90,6 +90,23 @@ Multiple TiFlash nodes elect a master to add or delete placement rules to PD, an
...
+    ## logging files.
+    # redact_info_log = false
+
+    ## Path of file that contains list of trusted SSL CAs. if set, the following settings
+    ## `cert_path` and `key_path` shouldn't be empty
```
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:104]:
```suggestion
    ## Path of the file that contains a list of trusted SSL CAs. If set, the following settings
    ## `cert_path` and `key_path` are also needed.
```

File: tiflash/tiflash-configuration.md [hunk 5]
```diff
@@ -90,6 +90,23 @@ Multiple TiFlash nodes elect a master to add or delete placement rules to PD, an
...
+    # ca_path = "/path/to/ca.pem"
+    ## Path of file that contains X509 certificate in PEM format.
```
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:106]:
```suggestion
    ## Path of the file that contains X509 certificate in PEM format.
```

File: tiflash/tiflash-configuration.md [hunk 6]
```diff
@@ -90,6 +90,23 @@ Multiple TiFlash nodes elect a master to add or delete placement rules to PD, an
...
+    # cert_path = "/path/to/tiflash-server.pem"
+    ## Path of file that contains X509 key in PEM format.
```
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:108]:
```suggestion
    ## Path of the file that contains X509 key in PEM format.
```

File: enable-tls-between-components.md [hunk 7]
```diff
@@ -61,11 +64,12 @@ Currently, it is not supported to only enable encrypted transmission of some spe
 
         ```toml
         [security]
-        # Path of file that contains list of trusted SSL CAs. If set, following four settings shouldn't be empty
+        ## The path for certificates. Empty string means that secure connections are disabled.
+        # Path of the file that contains list of trusted SSL CAs. If set, the following settings `cert_path` and `key_path` are also needed.
         cacert-path = "/path/to/ca.pem"
```
Comment (JaySon-Huang) [inline enable-tls-between-components.md:69]: Is `cacert-path` a typo or by design? @nolouch

Comment (JaySon-Huang) [reply enable-tls-between-components.md:69]: Confirm that it is not a typo.

**Comment by TomShawn** [comment]
@JaySon-Huang Please resolve the conflict, thanks~

File: tiflash/tiflash-configuration.md [hunk 8]
```diff
@@ -144,7 +144,24 @@ delta_index_cache_size = 0
...
-    max_memory_usage_for_all_queries = 0 
+    max_memory_usage_for_all_queries = 0
+
+## Security settings take effect starting from v4.0.5.
+[security]
+    ## This configuration item enables or disables log redaction. If the configuration value
+    ## is set to `true`, all user data in the log will be replaced by `?`.
+    ## Note that you also need to set `security.redact-info-log` for tiflash-learner's logging
+    ## in tiflash-learner.toml
+    ## logging files.
```
Comment (JaySon-Huang) [inline tiflash/tiflash-configuration.md:155]:
```suggestion
```

File: enable-tls-between-components.md [hunk 9]
```diff
@@ -49,9 +49,12 @@ Currently, it is not supported to only enable encrypted transmission of some spe
 
         ```toml
         [security]
-        # set the path for certificates. Empty string means disabling secure connections.
+        ## The path for certificates. Empty string means that secure connections are disabled.
+        # Path of the file that contains list of trusted SSL CAs. If set, the following settings `cert_path` and `key_path` are also needed.
```
Comment (TomShawn) [inline enable-tls-between-components.md:53]: @lysu PTAL the description changes in this file, thanks~

Comment (lysu) [reply enable-tls-between-components.md:53]: LGTM for those description

**Comment by JaySon-Huang** [comment]
@TomShawn Can we merge this PR?

File: enable-tls-between-components.md (hunk 9 above)
Comment (TomShawn) [inline enable-tls-between-components.md:53]:
```suggestion
        # Path of the file that contains a list of trusted SSL CAs. If it is set, the following settings `cert_path` and `key_path` are also needed.
```

File: enable-tls-between-components.md [hunk 10]
```diff
@@ -61,11 +64,12 @@ Currently, it is not supported to only enable encrypted transmission of some spe
 
         ```toml
         [security]
-        # Path of file that contains list of trusted SSL CAs. If set, following four settings shouldn't be empty
+        ## The path for certificates. Empty string means that secure connections are disabled.
+        # Path of the file that contains list of trusted SSL CAs. If set, the following settings `cert_path` and `key_path` are also needed.
```
Comment (TomShawn) [inline enable-tls-between-components.md:68]:
```suggestion
        # Path of the file that contains a list of trusted SSL CAs. If it is set, the following settings `cert_path` and `key_path` are also needed.
```

File: enable-tls-between-components.md [hunk 11]
```diff
@@ -75,21 +79,24 @@ Currently, it is not supported to only enable encrypted transmission of some spe
 
          ```toml
         [security]
-        # Path of file that contains list of trusted SSL CAs. if set, following four settings shouldn't be empty
+        ## The path for certificates. Empty string means that secure connections are disabled.
+        # Path of the file that contains list of trusted SSL CAs. If set, the following settings `cert_path` and `key_path` are also needed.
```
Comment (TomShawn) [inline enable-tls-between-components.md:83]:
```suggestion
        # Path of the file that contains a list of trusted SSL CAs. If it is set, the following settings `cert_path` and `key_path` are also needed.
```

File: enable-tls-between-components.md [hunk 12]
```diff
@@ -75,21 +79,24 @@ Currently, it is not supported to only enable encrypted transmission of some spe
...
+        # Path of the file that contains X509 key in PEM format.
         key_path = "/path/to/tiflash-server-key.pem"
         ```
 
         Configure in the `tiflash-learner.toml` file:
 
         ```toml
         [security]
-        # Sets the path for certificates. The empty string means that secure connections are disabled.
+        # Path of the file that contains list of trusted SSL CAs. If set, the following settings `cert_path` and `key_path` are also needed.
```
Comment (TomShawn) [inline enable-tls-between-components.md:95]:
```suggestion
        # Path of the file that contains a list of trusted SSL CAs. If it is set, the following settings `cert_path` and `key_path` are also needed.
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by TomShawn** [comment]
Follow-Up: align the description changes in `enable-tls-between-components.md` to Chinese.

File: enable-tls-between-components.md [hunk 13]
```diff
@@ -61,11 +64,12 @@ Currently, it is not supported to only enable encrypted transmission of some spe
 
         ```toml
         [security]
-        # Path of file that contains list of trusted SSL CAs. If set, following four settings shouldn't be empty
+        ## The path for certificates. Empty string means that secure connections are disabled.
```
Comment (TomShawn) [inline enable-tls-between-components.md:67]:
```suggestion
        ## The path for certificates. Empty string means that secure connections are disabled. 
```

File: enable-tls-between-components.md [hunk 14]
```diff
@@ -49,9 +49,12 @@ Currently, it is not supported to only enable encrypted transmission of some spe
 
         ```toml
         [security]
-        # set the path for certificates. Empty string means disabling secure connections.
+        ## The path for certificates. Empty string means that secure connections are disabled.
```
Comment (TomShawn) [inline enable-tls-between-components.md:52]:
```suggestion
        ## The path for certificates. An empty string means that secure connections are disabled.
```

File: enable-tls-between-components.md [hunk 15]
```diff
@@ -75,21 +79,24 @@ Currently, it is not supported to only enable encrypted transmission of some spe
 
          ```toml
         [security]
-        # Path of file that contains list of trusted SSL CAs. if set, following four settings shouldn't be empty
+        ## The path for certificates. Empty string means that secure connections are disabled.
```
Comment (TomShawn) [inline enable-tls-between-components.md:82]:
```suggestion
        ## The path for certificates. An empty string means that secure connections are disabled.
```

**Pull request merged**